specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,963 @@
+
+Thus, any backend that supports capabilities 1-4 and as many of capabilities
+5-9 as it desires is considered as supporting adaptive profile programs.
+An adaptive profile program must indicate what additional capabilities it uses
+via module flags. Additionally, backends can indicate how they support the
+various optional capabilities and any caveats on support for the capabilities
+that they do support via this [document](./Adaptive_Hardware/providers.md#backend-support-for-adaptive-profile).
+Ideally, static analysis/verification tools should be able to understand what
+capabilities of the adaptive profile a backend is implementing and should run
+a verification pass to ensure adaptive profile programs that are using capabilities
+not supported by a backend are rejected with an informative message.
+More details about each of the aforementioned capabilities are outlined below.
+
+## Mandatory Capabilities
+
+### Bullet 1: Quantum transformations
+
+The set of available instructions that transform the quantum state may vary
+depending on the targeted backend. The profile specification defines how to
+leverage and combine the available instructions to express a program, but does
+not dictate which quantum instructions may be used. Targeting a program to a
+specific backend requires choosing a suitable profile and quantum instruction
+set (QIS). Both can be chosen largely independently, though certain instruction
+sets may be incompatible with this (or other) profile(s). The section on the
+[quantum instruction set](#quantum-instruction-set) defines the requirements for
+a QIS to be compatible with the adaptive profile.
+More information about the role of
+the QIS, recommendations for front- and backend providers, as well as the
+distinction between run-time functions and quantum instructions, can be found in
+this [document](../Instruction_Set.md).
+
+### Bullet 2: Measurements
+
+The second requirement relieves a restriction from the base profile that qubits
+can only be measured at the end of the program and that no quantum operations
+can be performed on a measured qubit.
+The only requirement here is that all qubits must be measurable, but
+there are no restrictions on when these measurements are made, and there is
+no restriction necessitating that all qubits be measured. I.E. it's perfectly
+valid to measure a subset of qubits used in the program, not have all measured
+qubits in the output, and perform the measurements at different points.
+A function to turn measurement results into classical data, mid-circuit, is
+critical to using forward branching and conditional quantum execution.
+An example of such a function (`@_quantum__rt__read_result__body`) is below:
+
+```llvm
+%0 = tail call i1 @__quantum__rt__read_result__body(%Result* null)
+ br i1 %0, label %then, label %continue
+```
+
+### Bullet 3: Program output
+
+There is no change to output specifications with respect to the base profile, sans for the 
```

Comment:
Error codes are no longer present in the spec. Non-zero return values indicate an error has occurred, but are all provider/backend specific.